demos/code/Blocks/Chapter 5 - Functions.js

```diff
@@ -765,6 +602,480 @@ Blockly.C['func_return'] = function(block) {
+							if(this.getVar_ === ptr.funcParam_[i][3]){
+								this.isConst_ = ptr.funcParam_[i][0];
+								this.typegetVar_ = ptr.funcParam_[i][1];
+								this.ptrType_ = ptr.funcParam_[i][2];
+								this.isInitialized_ = ptr.funcParam_[i][4];
+							}
+						}
+					}
+					break;
+				case 'isStruct':
+					for(var i = 0; i < ptr.classVar_.length; ++i){
+						if(this.getVar_ === ptr.classVar_[i][3]){
+							this.isConst_ = ptr.classVar_[i][0];
+							this.typegetVar_ = ptr.classVar_[i][1];
+							this.ptrType_ = ptr.classVar_[i][2];
+							this.isInClass_ = true;
+						}
+					}
+					break;
+				case 'isClass':
+					break;
+			}
+
+			ptr = ptr.getSurroundParent();
+		}
+
+		//Set typegetVar_
+		ptr = this.parentBlock_;
+		while(ptr){
+
+			switch(ptr.getDataStr()){
+				case 'isVar':
+
+					//Stream data from var declaration block
+					if(this.getVar_ === ptr.getVar_){
+
+						this.typegetVar_ = ptr.typegetVar_;
+						//stream input value
+						this.value_ = ptr.value_;
+
+						this.isInitialized_ = ptr.isInitialized_;
+
+						this.isNull_ = ptr.isNull_;
+
+						this.ptrType_ = ptr.ptrType_;
+
+						//stream const option
+						this.isConst_ = ptr.isConst_;
+
+						return;
+					}
+
+					break;
+			}
+
+
+
+			ptr = ptr.parentBlock_;
+		}
+	},
+
+	allocateDropdown: function(){
+		//Delete any repeating elements in the 2d array
+		var temp = [];
+		var temp2 = [];
+
+		//grab the data from the matrix
+		for(var i = 0; i < this.paramNames_.length; ++i){
+			temp.push(this.paramNames_[i][1]);
+		}
+
+		this.paramNames_ = [];
+
+		for(var i = 0; i < temp.length; ++i){
+			if(temp2.indexOf(temp[i]) == -1){
+				temp2.push(temp[i]);
+			}
+		}
+
+		for(var i = 0; i < temp2.length; ++i){
+			this.paramNames_.push([temp2[i], temp2[i]]);
+		}
+
+		return this.paramNames_;
+	},
+
+	allocateWarnings: function(){
+		var TT = "";
+		C = C_Logic;
+
+		if(!this.parentBlock_){
+			TT += "Block warning, this block has a return and must be connected.\n";
+		}
+
+		//Errors if a variable exists
+		if(this.getVar_.length > 0){
+
+			var currentVarFound = false;
+
+			for(var i = 1; i < this.paramNames_.length; ++i){
+				if(this.getFieldValue('VAR') === this.paramNames_[i][1]){
+					currentVarFound = true;
+					break;
+				}
+			}
+
+			if(this.isInClass_){
+				let ptr = this.getSurroundParent();
+
+				while(ptr){
+					switch(ptr.getDataStr()){
+						case 'isStruct':
+
+							for(var i = 0; i < ptr.classVar_.length; ++i){
+								if(this.getVar_ === ptr.classVar_[i][3]){
+									currentVarFound = true;
+									break;
+								}
+							}
+
+							break;
+					}
+					ptr = ptr.getSurroundParent();
+				}
+
+			}
+
+
+			if(!currentVarFound){
+				TT += 'Block warning, variable "' + this.getVar_ + '" is not declared in this scope.\n';
+			}
+		}
+
+		if(TT.length > 0){
+			this.setWarningText(TT);
+		}
+		else {
+			this.setWarningText(null);
+		}
+	}
+
+};
+Blockly.C['get_function'] = function(block) {
```

Comment:
This is just a rename of the block. The reference in index.html has been updated as well.